app/main.py

```diff
@@ -1,3 +1,55 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float | Distance) -> Distance:
+        if not isinstance(other, Distance):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: int | float | Distance) -> Distance:
+        if not isinstance(other, Distance):
+            self.km += other
+            return self
+        self.km += other.km
+        return self
+
+    def __mul__(self, other: int | float | Distance) -> Distance:
+        return Distance(km=self.km * other)
+
+    def __truediv__(self, other: int | float | Distance) -> Distance:
+        return Distance(round(self.km / other, 2))
```

Comment:
It can't be Distance here. Your logic allows to have only int or float. Fix the type hint, please.